agent/bench-scripts/pbench-specjbb2005

```diff
@@ -325,7 +324,7 @@ specjbb_opts="-cp ${specjbb_cp} -Xms${heap_size} -Xmx${heap_size} ${java_opts} s
 for i in `seq $nr_jvms -1 1`; do
 	if [ $i -eq 1 ]; then
 		# the last JVM gets output so we can trigger the tools starting at iteration 1
-		java ${specjbb_opts} -id $i | tee ./jvm-$i-worker.txt | pbench-tool-trigger 1 "${benchmark_run_dir}" "${tool_group}" "${sample}"
+		java ${specjbb_opts} -id $i | tee ./jvm-$i-worker.txt | pbench-tool-trigger 1 "${benchmark_run_dir}" "${tool_group}" 1
```

Comment:
Arguably, we should replace `${sample}` with `$i`.  Replacing it with `1` instead is OK, because we know that, at this point, `$i` _is_ `1`, but, on that basis, should we replace the other `$i` (or `$i`'s) with `1`, as well?

(It also occurs to me that perhaps line 327 should be pulled out of the loop and executed afterwards, with the loop ending at `2` instead of `1`:  this would remove the need for the conditional, and it would better justify using a constant value instead of `$i`.)

Reply:
Pulling it out of the loop and executing it afterwards is a good idea.

But the use of `${i}` instead of `1` is actually incorrect from the meaning of the value of `i` verses the meaning of the 4th argument of `pbench-tool-trigger`.

The value of `i` is the ID of the worker.  The 4th argument is "which sample number of the given iteration should the tool trigger record under."

For `pbench-specjbb2005`, the `specjbb2005` benchmark is invoked once, and the last JVM's output is processed to determine when the benchmark changes between iterations.  So the fourth parameter should be dropped entirely since it does not need to be provided.  This is also how `pbench-user-benchmark` leverages `pbench-tool-trigger`.

In contrast, `pbench-trafficgen` invokes the benchmark (`trafficgen`) once per sample of an iteration, where it uses `pbench-tool-trigger` to determine where in the potentially long benchmark invocation should performance metrics be captured.  In that case, the iteration number and sample number need to be provided to `pbench-tool-trigger` to have the data recorded appropriately.

Reply:
Y'see, Peter...this is why we are so desperate to keep you around!